### ISSUE TITLE
Pay elsewhere button does not show up for Admin after report approval

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#37867]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
issue was found when executing #36814 
**Version Number:** 1.4.48-0
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** applause internal team
**Slack conversation:**

## Action Performed:
Precondition:
- The Collect workspace has admin and employee. Admin is also approver.
- The Collect workspace has indirect reimbursement.

1. [Employee] Go to workspace chat.
2. {Employee] Create a manual request.
3. [Admin] Go to workspace chat with employee.
4. [Admin] Click Approve.

## Expected Result:
Pay elsewhere button will show up for Admin after report approval.

## Actual Result:
Pay elsewhere button does not show up for Admin after report approval.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/66e04129-c89d-410a-bfe5-1ba3cfe8b33c

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015f3e017244f4a846</li>
        <li>Upwork Job ID: 1767788547695489024</li>
        <li>Last Price Increase: 2024-03-13</li>
    </ul>
</details>

**Comment by m-natarajan** [comment]
@MariaHCD FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by m-natarajan** [comment]
We think that this bug might be related to #wave7-collect-approvers
cc @RachCHopkins

**Comment by c3024** [comment]
Now we have the logic of checking if the user is a payer based on the reimbursement choice.

https://github.com/Expensify/App/blob/ffa731a28a2d1402148eef54e51d86fbb64321fe/src/libs/ReportUtils.ts#L1243-L1247

Backend sometimes sends the reimbursement choice blank. Only after changing it to `None` and back to `Indirect` as in the video does the reimbursement choice gets updated. Then the `Pay elsewhere` button appears fine.


https://github.com/Expensify/App/assets/102477862/6b13232f-d7c3-47a4-b096-061a8cb25aba

**Comment by MariaHCD** [comment]
@marcochavezf I see that we started sending reimbursementChoice here: https://github.com/Expensify/Web-Expensify/pull/40396

Looks like [getPolicyOnyxUpdates](https://github.com/Expensify/Web-Expensify/blob/5c1002bdfdc6dc090d5f64e411f53ae422b3a60a/lib/PolicyAPI.php#L6346-L6349) is deprecated now.

So perhaps we need to set reimbursementChoice in [fetchPolicies](https://github.com/Expensify/Web-Expensify/blob/d4ca9d6dcb640fea62708d0810adfeb4db149f74/lib/AppInit.php#L68-L76)?

I don't think this has to block the app deploy since it looks like a backend issue for now.

**Comment by marcochavezf** [comment]
We should be sending the reimbursement values from Auth [here](https://github.com/Expensify/Auth/blob/36da78949afb4774cfab6541bcac3353c30e1d51/auth/lib/Policy.cpp#L3727-L3728) (which is the function where we should use in favor of the deprecated getPolicyOnyxUpdates). I'm not sure if we should include it in the `fetchPolicies` since it gets the data from the policy summary and I think the data should be lean.

I'm wondering if we're not sending the values from Auth for this policy because it's not a primary policy? 🤔

**Comment by sakluger** [comment]
@marcochavezf @MariaHCD - just confirming that this a wave-collect bug, right? If so, can it be worked on externally, or does it need to be internal?

**Comment by MariaHCD** [comment]
I think it is a wave-collect bug, although I'll defer to @marcochavezf to confirm. As far as I can tell right now, this needs an internal fix and cannot be worked on externally.

**Comment by marcochavezf** [comment]
Yup, it should be part of wave-collect and agree it should be internal

**Comment by marcochavezf** [comment]
Hmm, a lot of automated tests are broken in the PR, it seems we'd need to come up with a different strategy.

**Comment by marcochavezf** [comment]
New PR up

**Comment by marcochavezf** [comment]
PR was deployed to production, it works now on both staging and production:

<img width="1149" alt="Screenshot 2024-03-18 at 2 56 32 p m" src="https://github.com/Expensify/App/assets/6829422/67dc8dcf-4efd-4d4b-8013-70a063541b79">
